daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -246,7 +250,10 @@ object SValue {
   }
 
   def toList(entries: TreeMap[SValue, SValue]): SList =
-    SList(FrontStack(entries.iterator.map { case (k, v) => entry(k, v) }.to[ImmArray]))
+    SList(
+      FrontStack(implicitly[Factory[SStruct, ImmArray[SStruct]]].fromSpecific(entries.iterator.map {
+        case (k, v) => entry(k, v)
+      })))
```

Comment:
We can support `.to(ImmArray)` by

1. adding an implicit conversion from `ImmArray.type` to `CanBuildFrom` to one of `object ImmArray`'s superclasses' companions for 2.12, and
2. duplicating the implicit conversion that makes stdlib companions implicitly convert to `to`'s argument for 2.13.

This won't work for the `TreeMap` case because the type arg can't be inferred in 2.12's form, but supporting it in this case should make porting nicer.

Reply:
Great idea, I’ve added it for ImmArray and fixed up the almost working one for `FrontStack` and switched over the cases in this PR and similar patterns from the first few porting PRs.